tests/loggers/test_mlflow.py

```diff
@@ -181,6 +181,46 @@ def training_epoch_end(self, *args, **kwargs):
     assert os.listdir(trainer.checkpoint_callback.dirpath) == [f"epoch=0-step={limit_batches - 1}.ckpt"]
 
 
+@mock.patch("pytorch_lightning.loggers.mlflow.mlflow")
+@mock.patch("pytorch_lightning.loggers.mlflow.MlflowClient")
+def test_mlflow_log_model(client, mlflow, tmpdir):
+    """Test that the logger creates the folders and files in the right place."""
+
+    mlflow.run = None
+    model = BoringModel()
+
+    # test log_model=True
+    logger = MLFlowLogger(experiment_name="exp-id-1", run_name="run-name-1", log_model=True)
```

Comment:
can you parametrize the log_model and use if-else statements for required assertions?

Reply:
I am not sure if I understood how, like the following?
<details>
  <summary markdown="span"

 `test_mlflow_log_model(client, mlflow, tmpdir, log_model):` 

</summary>

```
@mock.patch("pytorch_lightning.loggers.mlflow.mlflow")
@mock.patch("pytorch_lightning.loggers.mlflow.MlflowClient")
def test_mlflow_log_model(client, mlflow, tmpdir, log_model):
    """Test that the logger creates the folders and files in the right place."""

    model = BoringModel()

    client.return_value.log_artifact.reset_mock()
    client.return_value.log_artifacts.reset_mock()
    logger = MLFlowLogger(experiment_name="exp-id-1", run_name="run-name-1", log_model=log_model)
    trainer = Trainer(default_root_dir=tmpdir, logger=logger, max_epochs=2, limit_train_batches=3, limit_val_batches=3)
    trainer.fit(model)

    if log_model == True:
        # Checkpoint log
        client.return_value.log_artifact.assert_called_once()
        # Metadata and aliases log
        client.return_value.log_artifacts.assert_called_once()
    elif log_model == 'all':
            # Checkpoint log
            assert client.return_value.log_artifact.call_count == 2
            # Metadata and aliases log
            assert client.return_value.log_artifacts.call_count == 2
    elif log_model == False:
            # Checkpoint log
            assert not client.return_value.log_artifact.called
            # Metadata and aliases log
            assert not client.return_value.log_artifacts.called
```

</details>

I am not very familiar with tests, should I use MagicMock to instantiate log_model or should the function just be called 3 times (once per different value of `log_model`?

Reply:
yeah something similar. https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions

Reply:
Added this in the last commits, I modified it also for test_wandb.py